### ISSUE TITLE
Possible to use dimAmount with ModalDialogService in Angular?

### ISSUE THREAD

**Issue by FranciZ** [mukaschultze/nativescript-windowed-modal#6]
Is it possible to set the `dimAmount` in Angular through `ModalDialogeService.showModal` method?

The example in readme for Angular is opening modal through the `Page` reference but I've never really seen it done like this in NativeScript Angular:
```typescript
mainPage.showModal("./modal", {
  context: "I'm the context",
  closeCallback: (response: string) => console.log("Modal response: " + response),
  dimAmount: 0.5 // Sets the alpha of the background dim
} as ExtendedShowModalOptions)
```

Is there a way to do the same when opening modals with the `showModal` method on the `ModalDialogeService` like below? The `dimAmount` is not respected in this case right now.

```typescript
const options = {
  viewContainerRef: this.vcRef,
  fullscreen: false,
  context: {
    achievement
  },
  dimAmount: 1
};

this.modalService.showModal(ModalAchievementComponent, options);
```

**Comment by mukaschultze** [comment]
That should work fine actually, try casting the options with `as ExtendedShowModalOptions`, like this:

```ts
const options = {
  viewContainerRef: this.vcRef,
  fullscreen: false,
  context: {
    achievement
  },
  dimAmount: 1
} as ExtendedShowModalOptions;

this.modalService.showModal(ModalAchievementComponent, options);
```

I'll do some tests if it still doesn't work after that

**Comment by FranciZ** [comment]
Thanks but it doesn't work. In fact `showModal` expects `ModalDialogOptions` type so `ExtendedShowModalOptions` isn't accepted.

**Comment by jnorkus** [comment]
I would be glad to see this work too! The problem is not only the type mismatch. Here is the signature:
`showModal(type: Type<any>, { viewContainerRef, moduleRef, context, fullscreen, animated, stretched }: ModalDialogOptions): Promise<any>;`

This means the `dimAmount` will not be copied over to the options that windowed-modal can see.

**Comment by edusperoni** [comment]
This is an issue with nativescript-angular that **should** be addressed shortly, as it's now part of NS 5.2 release:

https://github.com/NativeScript/nativescript-angular/issues/1709#issuecomment-461956068

Since this would require a monkey patch of nativescript-angular, we're waiting for the core team to implement their solution so we can properly adapt.

**Comment by mukaschultze** [comment]
I'm closing this issue as the NativeScript/nativescript-angular#1771 PR was merged fixing this, the examples above should work fine now.
